agent.py

```diff
@@ -229,19 +229,16 @@ def doDeath(self, causeOfDeath):
 
     def doDisease(self):
         diseases = self.diseases
+        random.shuffle(diseases)
         for diseaseRecord in diseases:
             diseaseTags = diseaseRecord["disease"].tags
-            start = diseaseRecord["startIndex"]
-            end = diseaseRecord["endIndex"] + 1
             immuneResponse = [self.immuneSystem[i] for i in range(diseaseRecord["startIndex"], diseaseRecord["endIndex"] + 1)]
```

Comment:
Done. Hamming distance isn't really defined for strings of unequal lengths. We just iterate over the immune system and truncate any part of the disease that's outside of it to the right. When the disease is longer than the immune system, we're not checking if the right side of the disease tags might result in a closer Hamming distance if we truncate the left side instead. Perhaps it would be more interesting to slide whichever string is smaller over the window of the larger one? That way, if the disease is longer, you're checking all possible positions rather than only ones where the disease starts at immune system[0] or greater. I was working on this for a bit but it's big enough to be a separate PR. What are your thoughts?

Reply:
If I'm understanding it correctly, it shouldn't matter if we find a more optimal match as a substring. So long as the strings don't match, the disease never cures. In the case of a disease tag being longer than an immune system, finding a perfectly matching substring will have no effect on the resulting behavior.

Reply:
I guess that's true—I was thinking more about the theory of asymmetrical Hamming distance than the practice. To be fair, the specific tag being flipped might have an impact on other diseases the agent has.

Reply:
Good point from a computational perspective, but the immune systems don't have to be optimal much like real immune systems encountering a novel infection; they may be absolutely powerless to stop it.